simulator/protocols/interface.py

```diff
@@ -8,7 +8,7 @@ class IProtocol(ABC):
     provider: IProvider
 
     @classmethod
-    def instantiate(cls, provider: IProvider):
+    def instantiate(cls, provider: IProvider) -> 'IProtocol':
```

Comment:
Shouldn't this be without ' '

Reply:
Unfortunately the  IProtocol class is not available inside itself, since python declarations will be resolved at runtime. There is a class in the typing package called "Self" exactly for this use case, but it's really new, python 3.10 I think. This alternative also works for older python versions so I preferred using it